### PR TITLE
Add microsoft ouath2 guide

### PR DESCRIPTION
**Pull request by rtomas**
No image for the header yet.

### CODE REVIEW

**Comment by rtomas** [comment]
link to the guide: https://web3auth-docs-git-add-microsoft-ouath2-guide-web3auth.vercel.app/guides/microsoft-oauth

File: src/pages/guides/microsoft-oauth.mdx [hunk 1]
```diff
@@ -0,0 +1,188 @@
+---
+title: Integrating Microsoft OAuth with Web3Auth for Decentralized Authentication
+image: "content-hub/guides/banners/"
```
Comment (shahbaz17) [inline src/pages/guides/microsoft-oauth.mdx:3]: Use the image I sent you over DM here. Also on Auth Provider page of Microsoft

Comment (shahbaz17) [inline src/pages/guides/microsoft-oauth.mdx:3]: Use the image I sent you over DM here. Also on Auth Provider page of Microsoft

File: src/pages/guides/microsoft-oauth.mdx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+description: Learn how to integrate Microsoft Github OAuth with Web3auth
+type: guide
+tags: [microsoft, oauth, decentralized, authentication, github, web3]
+date: Febreaury 28, 2024
+author: Web3Auth Team
+order: 1
```
Comment (shahbaz17) [inline src/pages/guides/microsoft-oauth.mdx:9]: Please, make it like 3 or 4, so we have the current pinned one at the top.

File: src/pages/guides/microsoft-oauth.mdx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+communityPortalTopicId:
+---
+
+import SEO from "@site/src/components/SEO";
+
+<SEO
+  title="Integrating Microsoft OAuth with Web3Auth for Decentralized Authentication"
+  description="Learn how to integrate Microsoft Github OAuth with Web3auth."
+  image="https://web3auth.io/docs/content-hub/guides/banners/"
```
Comment (shahbaz17) [inline src/pages/guides/microsoft-oauth.mdx:18]: Add image here too.

**Review by shahbaz17 — CHANGES_REQUESTED** [review]
Overall looks good to me.

Please resolve the comments and then this is good to merge.

File: src/pages/guides/microsoft-oauth.mdx [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  slug="/content-hub/guides/microsoft-github-oauth"
+/>
+
+This guide will cover the basics of how to integrate Web3Auth with OAuth 2.0, using Microsoft Github authentication. Trying to explain all the flow
+and covering all the problems that can happen during the integration. Web3Auth is employed to provide Ethereum private key access and public address.
+
+Repository: https://github.com/rtomas/microsoft-oauth-for-web3auth
```
Comment (shahbaz17) [inline src/pages/guides/microsoft-oauth.mdx:25]: Also, let's bring this inside, `web3auth-pnp-examples` for now.

**Review by shahbaz17 — APPROVED** [review]
lgtm

**Pull request merged**